--- a/daemon/core/nodes/netclient.py
+++ b/daemon/core/nodes/netclient.py
@@ -176,7 +176,11 @@
         if netaddr.valid_ipv6(address.split("/")[0]):
             # IPv6 addresses are removed by default on interface down.
             # Make sure that the IPv6 address we add is not removed
-            self.run(f"{SYSCTL} -w net.ipv6.conf.{device}.keep_addr_on_down=1")
+            bridge_name = device.replace(".", "/")
+            try:
+                self.run(f"{SYSCTL} -w net.ipv6.conf.{bridge_name}.keep_addr_on_down=1")
+            except Exception:
+                pass
 
     def delete_address(self, device: str, address: str) -> None:
         """
